pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactedTopicImpl.java

```diff
@@ -69,16 +69,14 @@ public CompactedTopicImpl(BookKeeper bk) {
 
     @Override
     public CompletableFuture<CompactedTopicContext> newCompactedLedger(Position p, long compactedLedgerId) {
-        synchronized (this) {
```

Comment:
We can leave this lock here because JVM is smart enough to distinguish if we need it. or we can issue another PR to remove this lock to get a clear context. :)

Reply:
Ok.